client/ayon_resolve/api/plugin.py

```diff
@@ -901,6 +902,35 @@ def _create_parents(self):
             self.parents.append(parent)
 
 
+def get_editorial_publish_data(
+    folder_path,
+    product_name,
+    version=None
+) -> dict:
+    """Get editorial publish data from context.
+
+    Args:
+        folder_path (str): Folder path where editorial package is located.
+        product_name (str): Editorial product name.
+        version (Optional[str]): Editorial product version. Defaults to None.
+
+    Returns:
+        dict: Editorial publish data.
+    """
+    data = {
+        "id": AVALON_INSTANCE_ID,
+        "productType": "editorial_pkg",
+        "productName": product_name,
+        "folderPath": folder_path,
+        "active": True,
+    }
+
+    if version:
+        data["version"] = version
+
+    return data
```

Comment:
This feels a bit weird in `plugin.py`. Should this be in `lib.py` or in a dedicated module, e.g. `ayon_resolve.api.editorial` or whatever?

Reply:
This code is related to plugin related 'tool-set' so it seems to me it is ok to be added here. There is also `get_representation_files` in the plugin. This function is used by creator and loader plugin only.